### ISSUE TITLE
Check-in Variable needs to be nested

### ISSUE THREAD

**Issue by Davinkjohnson** [episphere/connect#387]
Currently: The check-in variable is a single variable for each participant

To be: The check-in variable must be a recurring item, nested under the visit or collection (TBD on exact nesting)

**Comment by anthonypetersen** [comment]
@brotzmanmj @Davinkjohnson should this one be on hold? it didn't seem like we had a clear understanding of how exactly this should be nested / how it would work in practice

**Comment by Davinkjohnson** [comment]
@anthonypetersen We'll need a minimum of a design solution for this before we launch. (But might need it resolved before launch, because otherwise we might have to repurpose this to "baseline check-in" or similar)

This wouldn't be "MVP" for the navigation testing though.

**Comment by brotzmanmj** [comment]
Added requirement: This also needs to have the following functionality: A participant can only be checked in once per visit. So one check in per baseline visit, one check-in per followup1 visit, etc.

**Comment by FrogGirl1123** [comment]
@brotzmanmj @anthonypeterson I am trying to look into this now (I can't log into GCP which is holding me up, trying to get a response from CBIIT).  I need to see how some of this is looking in prod as BioSpm_Visit_v1r0 (331584571) is a default variable, as such it might not be the best variable to nest under.  I think we are going to run into a situation similar to the refusal/withdrawal variables.

**Comment by FrogGirl1123** [comment]
@brotzmanmj @anthonypetersen @Davinkjohnson Good news! Other than the biospecimen incentive flags (which are in the participants table) none of the other biospecimen default variables are initialized when the participant record is created.  This gives us flexibility in setting up the nesting that we did not have for refusal/withdrawal since we don't have to try to nest around the default variables.  Because the clinical sites do not have a check-in variable, I suggest we nest under visit and test by adding a Follow-up visit to the options.  @anthonypetersen, I am assuming you are the backend person on this.  Could you please come up with a JSON schema/example file of how best to nest the biospecimen data under visit.  Be sure to consider what variables are repeated measures over time.  Once this is done we can meet to review.   
@brotzmanmj could you or someone on communications/biospecimens review the default biospecimen variables in the Master Data Dictionary (the copy you created for the biospecimens is fine) and indicate which defaults are needed for communication about biospecimens, then we'll figure out how best to work them in without disrupting the nesting.
